src/extensions/extension-discovery.ts

```diff
@@ -387,24 +391,19 @@ export class ExtensionDiscovery extends Singleton {
         await this.installPackage(extension.absolutePath);
       }
     }
-    const extensions = bundledExtensions.concat(userExtensions);
 
-    return this.extensions = new Map(extensions.map(extension => [extension.id, extension]));
+    return this.extensions = new Map(bundledExtensions.concat(userExtensions).map(extension => [extension.id, extension]));
   }
 
   /**
    * Write package.json to file system and install dependencies.
    */
-  async installBundledPackages(packageJsonPath: string, extensions: InstalledExtension[]) {
-    const packagesJson: PackageJson = {
-      dependencies: {}
-    };
-
-    extensions.forEach((extension) => {
-      packagesJson.dependencies[extension.manifest.name] = extension.absolutePath;
-    });
+  installBundledPackages(packageJsonPath: string, extensions: InstalledExtension[]) {
```

Comment:
so this does not be to be async since `extensionInstaller.installPackages()` returns promise and await is not needed here?

Reply:
That is true. An `async function` wraps the body in a promise and treats thrown errors as `reject`'s. But also, in JS there are no "nested promises". If you write `Promise.resolve(Promise.resolve(4))` you have a `Promise<number>`.

Reply:
I think we should add explicit return type here. It's not obvious now that `installBundledPackages` returns a promise.